spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -294,12 +338,20 @@ def set_spyder_breakpoints(self):
     def set_pdb_ignore_lib(self):
         """Set pdb_ignore_lib into a debugging session"""
         self.call_kernel(interrupt=True).set_pdb_ignore_lib(
-            CONF.get('run', 'pdb_ignore_lib', False))
+            CONF.get('ipython_console', 'pdb_ignore_lib', False))
```

Comment:
You need to rewrite this method as it appears below, so you can apply our Preferences in the way I mentioned above:

```python
def set_pdb_ignore_lib(self, state):
    """Set pdb_ignore_lib into a debugging session"""
    self.call_kernel(interrupt=True).set_pdb_ignore_lib(state)
```

Reply:
The same goes for `set_pdb_execute_events` and `set_pdb_use_exclamation_mark` below.